src/components/LoginForm.js

```diff
@@ -0,0 +1,79 @@
+import React, { useState } from 'react'
+import { Link, useHistory } from 'react-router-dom'
+import { isUserValid } from '../utils/authentication'
+
+export default function LoginForm() {
+    const [error, setError] = useState('')
+    const history = useHistory()
+    const [loginDetails, setLoginDetails] = useState({ username: '', password: '' })
+
+    const submitHandler = async (e) => {
+        e.preventDefault()
+
+        try {
+            const isValid = await isUserValid(loginDetails)
+            if (isValid) {
+                history.push({ pathname: "/welcome", state: { username: loginDetails.username } })
+            } else {
+                setError('INVALID CREDENTIALS')
+            }
+        } catch (err) {
+            setError('INVALID CREDENTIALS')
+        }
+    }
+
+    return (
+        <div className="container"
+            style={{
+                display: "flex", flexDirection: "column", height: "100vh",
+                justifyContent: "center", alignItems: "center"
+            }}>
+            <div className="card mt-5"
+                style={{ width: "500px", margin: "auto" }}>
+                <div className="card-body text-center">
+                    <form onSubmit={submitHandler} >
+
+                        <div className="mt-1">
+                            <h2 className="h3 font-weight-light">Sign into your account</h2>
+                        </div>
+
+                        {(error !== '') ? (<div className="alert-warning p-2 mt-3">{error}</div>) : ''}
+
+                        <div className="">
+                            <label htmlFor="username" className="sr-only">Username: </label><br />
+                            <input type="text" className="form-control" name="username" id="username"
+                                placeholder="username" required autoFocus
+                                onChange={(e) => setLoginDetails({ ...loginDetails, username: e.target.value })}
```

Comment:
"onChange" should be a handler defined as the "submitHandler" one. Avoid updating state by hooks directly in jsx.